### PR TITLE
Remove some files from archive

### PR DESCRIPTION
**Pull request by phated**
Found a neat trick to ignore some files from the generated archive using `.gitattributes` (anything in gitignore is already skipped so I added archives and unarchived directories into the ignore list).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,7 @@ node_modules/
 _esy/
 .merlin
 META
-.vscode/
\ No newline at end of file
+.vscode/
+.DS_Store
+*.tar.gz
+binaryen*
```
Comment (ospencer) [inline .gitignore:8]: Does this only hit files and not the directory?

Comment (phated) [reply .gitignore:8]: I'm pretty sure it only matches files at the root, but I can make it `/binaryen*/` if you want

Comment (ospencer) [reply .gitignore:8]: I think I was just confused. This is just to not include the opam file and any archives created manually?

Comment (phated) [reply .gitignore:8]: No, I think you actually caught an issue with this. While I was building the archives locally, I noticed that I left old untarred directories hanging around and I wanted to make sure they didn't end up in the other archives.

Comment (phated) [inline .gitignore:8]:
```suggestion

```

Comment (phated) [reply .gitignore]: I removed that line because it isn't really an issue with the new github action.

**Pull request merged**